app/models/manageiq/providers/lenovo/physical_infra_manager/parser/firmware_parser.rb

```diff
@@ -9,19 +9,67 @@ class << self
         :name         => :name
       }.freeze
 
+      FORMAT_BY_TYPE_ROLE = %w(UEFI IMM2 XCC).freeze
+      FORMAT_BY_NAME = %w(DRVWN LXPM DRVLN).freeze
+      FORMAT_BY_DISTINCT_NAME = {
+        'DSA' => 'Diagnostic'
+      }.freeze
+
+      RULES = {
+        :format_by_name          => FORMAT_BY_NAME,
+        :format_by_type_role     => FORMAT_BY_TYPE_ROLE,
+        :format_by_distinct_name => FORMAT_BY_DISTINCT_NAME
+      }.freeze
+
+      PARSER_FIRMWARE_TYPE = {
+        'UEFI-Backup' => 'UEFI',
+        'IMM2-Backup' => 'IMM2',
+        'XCC-Backup'  => 'XCC'
+      }.freeze
+
       #
       # Parses a firmware into a Hash
       #
       # @param [Hash] firmware - object containing details for the firmware
       #
       # @return [Hash] the firmware as required by the application
       #
-      def parse_firmware(firmware)
+      def parse_firmware(firmware, component = nil)
+        @component = component
         parse(firmware, FIRMWARE)
       end
 
       def name(firmware)
-        "#{firmware['role']} #{firmware['name']}-#{firmware['status']}".strip
+        if @component.present?
```

Comment:
You can combine both conditions using a safe navigation,
something like `@component&.[](:name)  == 'N/A' ? ...`

Reply:
done